backend/dal/Services/Concrete/PropertyService.cs

```diff
@@ -0,0 +1,146 @@
+        /// Will not return sensitive properties unless the user has the `sensitive-view` claim and belongs to the owning agency.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public Paged<Property> Get(ParcelFilter filter)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.PropertyView);
+
+            var query = this.Context.GenerateQuery(this.User, filter)
+                    .Include(p => p.Status)
+                    .Include(p => p.Classification)
+                    .Include(p => p.Agency)
+                    .Include(p => p.Address)
+                    .ThenInclude(a => a.City)
+                    .Include(p => p.Address)
+                    .ThenInclude(a => a.Province);
+
+            var total = query.Count();
+            var items = query.Select(p => new Property(p)
+            {
+                Status = p.Status.Name,
+                Classification = p.Classification.Name,
+                AddressId = p.AddressId,
+                Address = $"{p.Address.Address1} {p.Address.Address2}",
+                City = p.Address.City.Name,
+                Province = p.Address.Province.Name,
+                Agency = p.Agency.Name
+            }).Skip((filter.Page - 1) * filter.Quantity).Take(filter.Quantity);
+
+            return new Paged<Property>(items, filter.Page, filter.Quantity, total);
+        }
+
+        /// <summary>
+        /// Get an array of building properties within the specified filters.
+        /// Will not return sensitive properties unless the user has the `sensitive-view` claim and belongs to the owning agency.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public Paged<Property> Get(BuildingFilter filter)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.PropertyView);
+
+            var query = this.Context.GenerateQuery(this.User, filter)
+                .Include(b => b.Parcel)
+                .Include(b => b.Status)
+                .Include(b => b.Classification)
+                .Include(b => b.Agency)
+                .Include(b => b.Address)
+                .ThenInclude(a => a.City)
+                .Include(b => b.Address)
+                .ThenInclude(a => a.Province)
+                .Include(b => b.BuildingConstructionType)
+                .Include(b => b.BuildingPredominateUse)
+                .Include(b => b.BuildingOccupantType);
+
+            var total = query.Count();
+            var items = query.Select(b => new Property(b)
+            {
+                PID = b.Parcel.PID,
+                PIN = b.Parcel.PIN,
+                Status = b.Status.Name,
+                Classification = b.Classification.Name,
+                Agency = b.Agency.Name,
+                AddressId = b.AddressId,
+                Address = $"{b.Address.Address1} {b.Address.Address2}",
```

Comment:
Same comment here about extra `space`

Reply:
I can't do the trim or the query will not run in the database regrettably.